### PR TITLE
CIF-1437: Content Teaser CTA bug fix

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

The category URL is now rendered properly.
The author is not able anymore to set both category and product field on the content teaser actions.

## Related Issue

CIF-1437

## Motivation and Context

When the content teaser component is configured to link to a category page, the CTA button doesn't work. However, when the component is configured to link to a product page, the CTA button works. If the component is configured for both fields, the content teaser CTA button uses the category field

## How Has This Been Tested?

Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                categoryElement.off('change', handleCategoryChange);
+
+                const eventData = { productField, categoryField };
+                // [re]attach change handlers
+                productElement.on('change', eventData, handleProductChange);
+                categoryElement.on('change', eventData, handleCategoryChange);
+            });
+    };
+
+    // sets an empty value on the category field when product field gets updated
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js:81]: I have to admit I'm surprised there is so much code in this file, I naively thought we would only need (more or less) these two functions to implement the "XOR fields" (one or the other but not both), with only some "onChange" event handlers. Or what do I miss?

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java [hunk 2]
```diff
@@ -90,7 +90,13 @@ private String toUrl(SlingHttpServletRequest request, Page page, Map<String, Str
 
         StringSubstitutor sub = new StringSubstitutor(params, prefix, suffix);
         String url = sub.replace(template);
-        return StringUtils.substringBeforeLast(url, "#" + prefix); // remove anchor if it hasn't been substituted
+        url = StringUtils.substringBeforeLast(url, "#" + prefix); // remove anchor if it hasn't been substituted
+
+        if (url.contains("{{") || url.contains("${")) {
```
[the remark this answers is not in the document]
Comment (laurentiumagureanu) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/UrlProviderImpl.java:95]: I think removing all left-over variables might produce unexpected results for the user.
better to just log the warning and leave it as it is

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/teaser/CommerceTeaserActionItem.java [hunk 3]
```diff
@@ -60,7 +60,7 @@ public String getURL() {
             Map<String, String> params = new ParamsBuilder().urlKey(selector).map();
             return urlProvider.toProductUrl(request, page, params);
         } else {
-            Map<String, String> params = new ParamsBuilder().id(selector).map();
+            Map<String, String> params = new ParamsBuilder().id(selector).urlPath("").map();
```
[the remark this answers is not in the document]
Comment (laurentiumagureanu) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/teaser/CommerceTeaserActionItem.java:63]: see the above comment

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js (hunk 1 above)
Comment (laurentiumagureanu) [reply ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js:81]: the fact that the teaser allows for multiple actions makes it a little more complicated.
but maybe I missed something. let me know if there's a simpler way of doing this

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+ ******************************************************************************/
+
+($ => {
+    'use strict';
+
+    const dialogContentSelector = '[data-cmp-is="commerceteaser-editor"].cmp-teaser__editor';
+    const productFieldSelector = '[data-cmp-teaser-v1-dialog-edit-hook="actionLink"][placeholder="Product"]';
+    const categoryFieldSelector = '[data-cmp-teaser-v1-dialog-edit-hook="actionLink"][placeholder="Category"]';
+    const actionsMultifieldSelector = '.cmp-teaser__editor-multifield_actions';
+    const actionsMultifieldItemSelector = '.coral3-Multifield-item';
```
Comment (herzog31) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js:22]: Please check GRANITE-20375. You're not supposed to use any selectors with `.coral-*`, `.coral3-*` and `.granite-*`.

File: ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
```
Comment (herzog31) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js:1]: Please add a test for this file and make sure that https://github.com/adobe/aem-core-cif-components/blob/master/ui.apps/karma.conf.js includes it for coverage.

**Pull request merged**